src/test/java/cz/muni/fi/pa165/TrainerFightDaoTest.java

```diff
@@ -0,0 +1,275 @@
+import cz.muni.fi.pa165.kodemon.dao.TrainerFightDao;
+import cz.muni.fi.pa165.kodemon.dao.TrainerDao;
+import cz.muni.fi.pa165.kodemon.dao.GymDao;
+import cz.muni.fi.pa165.kodemon.entity.TrainerFight;
+import cz.muni.fi.pa165.kodemon.entity.Trainer;
+import cz.muni.fi.pa165.kodemon.entity.Gym;
+import cz.muni.fi.pa165.kodemon.enums.PokemonType;
+import org.springframework.dao.DataAccessException;
+import org.springframework.data.domain.Example;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+import org.springframework.transaction.annotation.Transactional;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+/**
+ * Unit tests of {@link TrainerFightDao}
+ *
+ * @author Miso Romanek
+ */
+@Transactional
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+@ContextConfiguration(classes = KodemonApplicationContext.class)
+public class TrainerFightDaoTest extends AbstractTestNGSpringContextTests {
+
+    @Inject
+    private TrainerFightDao trainerFightDao;
+
+    @Inject
+    private TrainerDao trainerDao;
+
+    @Inject
+    private GymDao gymDao;
+
+    private TrainerFight trainerFight;
+    private Trainer trainer;
+    private Gym gym;
+
+    @BeforeMethod
+    void prepare() {
+        assertThat(trainerFightDao, is(notNullValue(TrainerFightDao.class)));
+        trainerFightDao.deleteAll();
+        assertThat("trainerFightDao.count() != 0L", trainerFightDao.count(), is(0L));
+
+        assertThat(trainerDao, is(notNullValue(TrainerDao.class)));
+        trainerDao.deleteAll();
+        assertThat("trainerDao.count() != 0L", trainerDao.count(), is(0L));
+
+        assertThat(gymDao, is(notNullValue(GymDao.class)));
+        gymDao.deleteAll();
+        assertThat("gymDao.count() != 0L", gymDao.count(), is(0L));
+
+        prepareTrainer();
+        assertThat("trainer == null", trainer, is(notNullValue()));
+        trainerDao.save(trainer);
+
+        prepareGym();
+        assertThat("gym == null", gym, is(notNullValue()));
+        gymDao.save(gym);
+
+        trainerFight = null;
+    }
+
+    /** save(...) tests */
+
+    @Test(expectedExceptions = {DataAccessException.class})
+    void testSaveNullTrainerFight() {
+        trainerFight = null;
+        assertThat(trainerFight, is(nullValue(TrainerFight.class)));
+        trainerFightDao.save(trainerFight);
+    }
+
+    @Test
+    void testSaveCorrectInstance() {
+        trainerFight = randomTrainerFight(1);
+        trainerFight.setWasChallengerSuccessful(true);
```

Comment:
Naco sa tu vola toto, ked sa to vola v `randomTrainerFight()`? ved na to ta metoda je, ze generuje random trainer fight + hadze to NPE, lebo tam nenastavujes trainera ani gym, co su `@NotNull` fieldy